### ISSUE TITLE
Ignoring replies

### ISSUE THREAD

**Issue by probgoblin** [yogthos/mastodon-bot#8]
I have tried a variety of regular expressions to ignore any replies that start with @, but nothing seems to work. Is there an option I can use in the .cljs file to accomplish this?

**Comment by yogthos** [comment]
You shouldn't have to do anything special for that. For example, this looks to work as expected:

```clojure
cljs.user=> (def content-filter-regexes (mapv re-pattern [".*@.*"]))
#'cljs.user/content-filter-regexes
cljs.user=> (defn blocked-content? [text]
       #_=> (boolean (some #(re-matches % text) content-filter-regexes)))
#'cljs.user/blocked-content?
cljs.user=> (blocked-content? "foo")
false
cljs.user=> (blocked-content? "@foo")
true
cljs.user=> (blocked-content? "foo @bar baz")
true
```

**Comment by probgoblin** [comment]
Thank you!

I used that regular expression in the content filter section of the individual config files and it no works correctly (putting it directly into the .cljs had a weird impact on post length).

This issue is resolved.

Also, is there any chance of this supporting twitter threads?

**Comment by yogthos** [comment]
I haven't looked at parsing threads yet. It shouldn't be too difficult in principle, but I probably won't have time to dig into this in the near future. I've marked the issue as help wanted in case somebody gets around to it before I do.

**Comment by daquinons** [comment]
Yep, I'm trying to do a regex filter that leaves out the replies but just realized that it would leave out also the threads because they are just tweets replying to another self tweet, a regex that filters everything that starts with "@" will leave them out.

Also, if the regex filters anything with a mention no matter where the mention is, it will leave a lot of tweets out (for example, anything shared from a newspaper that appends "via @nameofsite").

**Comment by daquinons** [comment]
Ok! I must say that I have zero experience with Clojure, but I finally made it work. As @probgoblin stated, no regex pattern was working, all of them were returning false. After printing and trying with different ones I looked at clojure documentation and I found re-find.

> ;; The distinction is that re-find tries to find _any part_ of the string
> ;; that matches the pattern, but re-matches only matches if the _entire_
> ;; string matches the pattern.

```
(defn blocked-content? [text]
  (boolean (some #(re-find % text) content-filter-regexes)))

```

So this made the trick, at least with the few cases that I tried.

Also, I'm using this regex to remove replies only and not mentions in the middle of the tweet. Threads will still be removed with this hacky solution. Definitely not optimal (I would think on an option include-replies like include-rts, and threads would never be filtered no matter the option). Also, this regex will work even if you change the format of the toot (in my case I prefer to have the twitter account at the beginning, so just checking from the beginning of the string to look for @ would not work.

:content-filters ["(?m)^[@]"]}}

I hope it helps :)
![Image](https://i.imgur.com/tdRnwxx.png)
